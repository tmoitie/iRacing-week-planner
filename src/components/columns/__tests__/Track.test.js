import React from 'react';
import { render, screen } from '@testing-library/react';
import { describe, test, expect } from '@jest/globals';
import { TableWrapper } from './ColumnUtils';

import '@testing-library/jest-dom';

import { Track } from '../index';

describe('components/columns/Track', () => {
  test('renders no own or fave', () => {
    render(
      <TableWrapper>
        <Track
          race={{
            trackId: 125,
          }}
          ownedTracks={[]}
          favouriteTracks={[]}
        />
      </TableWrapper>,
    );

    expect(screen.getByTestId('table-row').firstChild).toMatchSnapshot();
    expect(screen.getByTestId('table-row').firstChild).not.toHaveClass('success');
    expect(screen.queryByTestId('StarIcon')).not.toBeInTheDocument();
  });

  test('renders ownership', () => {
    render(
      <TableWrapper>
        <Track
          race={{
            trackId: 125,
          }}
          ownedTracks={[125]}
          favouriteTracks={[]}
        />
      </TableWrapper>,
    );

    expect(screen.getByTestId('table-row').firstChild).toMatchSnapshot();
    expect(screen.getByTestId('table-row').firstChild).toHaveClass('success');
    expect(screen.queryByTestId('StarIcon')).not.toBeInTheDocument();
  });

  test('renders favourite', () => {
    render(
      <TableWrapper>
        <Track
          race={{
            trackId: 125,
          }}
          ownedTracks={[]}
          favouriteTracks={[125]}
        />
      </TableWrapper>,
    );

    expect(screen.getByTestId('table-row').firstChild).toMatchSnapshot();
    expect(screen.getByTestId('table-row').firstChild).not.toHaveClass('success');
    expect(screen.queryByTestId('StarIcon')).toBeInTheDocument();
  });
});
