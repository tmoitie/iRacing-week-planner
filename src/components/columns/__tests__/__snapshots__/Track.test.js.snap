// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/columns/Track renders favourite 1`] = `
<td
  class=""
>
  <div>
    <span
      class="glyphicon glyphicon-star"
      data-testid="StarIcon"
    />
    <span>
       
    </span>
  </div>
</td>
`;

exports[`components/columns/Track renders no own or fave 1`] = `
<td
  class=""
>
  <div>
    <span>
       
    </span>
  </div>
</td>
`;

exports[`components/columns/Track renders ownership 1`] = `
<td
  class="success"
>
  <div>
    <span>
       
    </span>
  </div>
</td>
`;
